domain/fhir/eoverdracht/converters.go

```diff
@@ -0,0 +1,122 @@
+		PatientProperties: types.PatientProperties{
+			Dob:       &types2.Date{Time: dob},
+			Email:     nil,
+			FirstName: p.Get(`name.0.given.0`).String(),
+			Gender:    gender,
+			Ssn:       &ssn,
+			Surname:   p.Get(`name.0.family`).String(),
+			Zipcode:   p.Get(`address.0.postalCode`).String(),
+		},
+		AvatarUrl: &avatar,
+	}
+}
+
+func FHIRAdvanceNoticeToDomainTransfer(notice AdvanceNotice) (types.TransferProperties, error) {
+	patient := FHIRPatientToDomainPatient(notice.Patient)
+	adminData, err := FilterCompositionSectionByType(notice.Composition.Section, AdministrativeDocCode)
+	if err != nil {
+		return types.TransferProperties{}, fmt.Errorf("administrativeData section missing in advance notice: %w", err)
+	}
+	transferDate, _ := time.Parse(time.RFC3339, *(*string)(adminData.Extension[0].ValueDateTime))
+
+	domainTransfer := types.TransferProperties{
+		Patient:      patient,
+		CarePlan:     types.CarePlan{PatientProblems: []types.PatientProblem{}},
+		TransferDate: types2.Date{Time: transferDate},
+	}
+
+	for _, condition := range notice.Problems {
+		var interventions []types.Intervention
+		for _, procedure := range notice.Interventions {
+			if fhir.FromStringPtr(procedure.ReasonReference[0].Reference) == "Condition/"+fhir.FromIDPtr(condition.ID) {
+				interventions = append(interventions, FHIRProcedureToDomainIntervention(procedure))
+			}
+		}
+		domainTransfer.CarePlan.PatientProblems = append(domainTransfer.CarePlan.PatientProblems,
+			types.PatientProblem{
+				Interventions: interventions,
+				Problem:       FHIRConditionToDomainProblem(condition),
+			})
+
+	}
+	return domainTransfer, nil
+}
+
+func FHIRNursingHandoffToDomainTransfer(nursingHandoff NursingHandoff) (types.TransferProperties, error) {
+	patient := FHIRPatientToDomainPatient(nursingHandoff.Patient)
+	adminData, err := FilterCompositionSectionByType(nursingHandoff.Composition.Section, AdministrativeDocCode)
+	if err != nil {
+		return types.TransferProperties{}, fmt.Errorf("administrativeData section missing in advance notice: %w", err)
+	}
+	transferDate, _ := time.Parse(time.RFC3339, *(*string)(adminData.Extension[0].ValueDateTime))
+
+	domainTransfer := types.TransferProperties{
+		Patient:      patient,
+		CarePlan:     types.CarePlan{PatientProblems: []types.PatientProblem{}},
+		TransferDate: types2.Date{Time: transferDate},
+	}
+
+	for _, condition := range nursingHandoff.Problems {
+		var interventions []types.Intervention
+		for _, procedure := range nursingHandoff.Interventions {
+			if fhir.FromStringPtr(procedure.ReasonReference[0].Reference) == "Condition/"+fhir.FromIDPtr(condition.ID) {
+				interventions = append(interventions, FHIRProcedureToDomainIntervention(procedure))
+			}
+		}
+		domainTransfer.CarePlan.PatientProblems = append(domainTransfer.CarePlan.PatientProblems,
+			types.PatientProblem{
+				Interventions: interventions,
+				Problem:       FHIRConditionToDomainProblem(condition),
+			})
+
+	}
+	return domainTransfer, nil
+}
```

Comment:
Wouldn't it make more sense to call these functions `ParseName` where `Name` is the name of the domain struct. The last function would become: `ParseTransferProperties`. It would make it a lot easier to read.

Reply:
Remove the FHIRName from method: `FHIRPatientToDomainPatient` -> `ToDomainPatient`